File: go-client-sdk/pkg/models/operations/getcustomersexternalcustomeridexternalcustomerid.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetCustomersExternalCustomerIDExternalCustomerIDPathParams struct {
	ExternalCustomerID string `pathParam:"style=simple,explode=false,name=external_customer_id"`
}

type GetCustomersExternalCustomerIDExternalCustomerIDRequest struct {
	PathParams GetCustomersExternalCustomerIDExternalCustomerIDPathParams
}

type GetCustomersExternalCustomerIDExternalCustomerIDResponse struct {
	ContentType string
	Customer    *shared.Customer
	StatusCode  int64
}

File: typescript-client-sdk/src/sdk/models/operations/getcustomersexternalcustomeridexternalcustomerid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetCustomersExternalCustomerIdExternalCustomerIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=external_customer_id" })
  externalCustomerId: string;
}


export class GetCustomersExternalCustomerIdExternalCustomerIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCustomersExternalCustomerIdExternalCustomerIdPathParams;
}


export class GetCustomersExternalCustomerIdExternalCustomerIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  customer?: shared.Customer;

  @SpeakeasyMetadata()
  statusCode: number;
}
